chain/network/src/peer_manager/connection/mod.rs

```diff
@@ -196,6 +200,8 @@ pub(crate) struct PoolSnapshot {
     /// Connections which have completed the handshake and are ready
     /// for transmitting messages.
     pub ready: im::HashMap<PeerId, Arc<Connection>>,
```

Comment:
Hmm... these two hashmaps plus the conditioned and defensive code used to manipulate them makes me uncomfortable. How about we still use a single map but the key of the map is an enum of either PeerId or PublicKey, and Connection has a connection_key() which returns that enum?

Reply:
I want the connection to be both accessible via PeerId and PublicKey, so I need 2 indices for that, or one index keyed by an enum but with multiple entries for a single Connection. The approaches are isomorphic and the amount of related code won't change either.